src/rgw/rgw_op.cc

```diff
@@ -1928,8 +1928,14 @@ void RGWCreateBucket::execute()
   if (!store->is_meta_master()) {
     JSONParser jp;
     op_ret = forward_request_to_master(s, NULL, store, in_data, &jp);
-    if (op_ret < 0)
+    if (op_ret < 0) {
+      if (op_ret == -EINVAL){
+	// Taking the easy way out, basically some issue in processing request from our side
+	// return a 5XX instead of an -EINVAL to the end user
+	op_ret = -ERR_SERVICE_UNAVAILABLE;
+      }
```

Comment:
@cbodley I've pushed another version where I've moved to up forward_request itself and see the value of process(), partly because if the master genuinely returned a -EINVAL, we shouldn't be reraising a 500